components/common/usage_percent_bar/index.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import styled from 'styled-components';
+
+type Thresholds = {
+    ok: number;
+    warn: number;
+    danger: number;
+    exceeded: number;
+}
+
+type Props = {
+    percent: number;
+    thresholds?: Thresholds;
+    barWidth?: number;
+}
+
+const defaultThresholds = {
+    ok: 0,
+    warn: 40,
+    danger: 80,
+    exceeded: 100.0000001,
+};
+
+type BarBackgroundProps = {
+    width: number;
+    thresholds: Thresholds;
+    percent: number;
+}
+
+const BarBackground = styled.div<BarBackgroundProps>`
+    height: ${(props) => Math.ceil(props.width / 20)}px;
+    width: ${(props) => props.width}px;
+    background: ${(props) => (
+        isExceeded(props.percent, props.thresholds) ?
+            'var(--dnd-indicator)' :
+            'rgba(var(--center-channel-text-rgb), 0.11)'
+    )};
+    border-radius: 8px;
+    position: relative;
+`;
+
+type BarForegroundProps = {
+    width: number;
+    percent: number;
+    thresholds: Thresholds;
+}
+
+function getColor(percent: number, thresholds: Thresholds): string {
+    if (percent >= thresholds.ok && percent < thresholds.warn) {
+        return 'var(--online-indicator)';
+    } else if (percent >= thresholds.warn && percent < thresholds.danger) {
+        return 'var(--away-indicator)';
+    } else if (percent >= thresholds.danger && percent < thresholds.exceeded) {
+        return 'var(--dnd-indicator)';
+    } else if (percent >= thresholds.exceeded) {
+        return 'var(--away-indicator)';
+    }
+    return '';
+}
+
+function isExceeded(percent: number, thresholds: Thresholds): boolean {
+    return percent >= thresholds.exceeded;
+}
+
+const BarForeground = styled.div<BarForegroundProps>`
+    height: 100%;
+    width: ${(props) => (isExceeded(props.percent, props.thresholds) ? 91 : props.percent)}%;
+    border-radius: 8px;
+    background-color: ${(props) => getColor(props.percent, props.thresholds)};
+    transition: background-color 0.4s ease, width 0.4s ease;
+    position: absolute;
+`;
+
+const UsagePercentBar = (props: Props) => {
```

Comment:
I'm hoping we can reuse this for the other places that display limits to the user outside the system console.

Reply:
Maybe a storybook could be helpful.

Reply:
It looks like storybook doesn't run or build locally right now. From the error, it looks related to the change that exports types as @mattermost/types. I will try again once I take the ticket to more fully implement the admin dashboard limits UI: https://mattermost.atlassian.net/browse/MM-43085 .